nibabies/workflows/bold/resampling.py

```diff
@@ -117,63 +156,518 @@ def select_target(subject_id, space):
+        iterfield=["in_file", "hemi"],
+        name="prepend_hemi",
+        mem_gb=DEFAULT_MEMORY_MIN_GB,
+        run_without_submitting=True,
+    )
+    prepend_hemi.inputs.hemi = ["lh", "rh"]
+
     update_metadata = pe.MapNode(
         GiftiSetAnatomicalStructure(),
         iterfield=["in_file"],
         name="update_metadata",
         mem_gb=DEFAULT_MEMORY_MIN_GB,
+        run_without_submitting=True,
     )
 
     outputnode = pe.JoinNode(
         niu.IdentityInterface(fields=["surfaces", "target"]),
         joinsource="itersource",
         name="outputnode",
+        run_without_submitting=True,
     )
 
+    if project_goodvoxels:
+        workflow, combine_ribbon_vol_hemis = make_ribbon_file(workflow, mem_gb, inputnode)
+        workflow, goodvoxels_mask, ribbon_boldsrc_xfm = mask_outliers(
+            workflow, mem_gb, rename_src, combine_ribbon_vol_hemis
+        )
+        workflow, apply_goodvoxels_ribbon_mask = project_vol_to_surf(
+            workflow, mem_gb, rename_src, goodvoxels_mask, ribbon_boldsrc_xfm
+        )
+
+        # If not applying goodvoxels, then get sampler.source_file from rename_src
+        # instead in connect_bold_surf_wf
+        apply_goodvx_or_rename = apply_goodvoxels_ribbon_mask
+    else:
+        apply_goodvx_or_rename = rename_src
+
     # fmt: off
-    workflow.connect([
-        (inputnode, get_fsnative, [('subject_id', 'subject_id'),
-                                   ('subjects_dir', 'subjects_dir')]),
-        (inputnode, targets, [('subject_id', 'subject_id')]),
-        (inputnode, rename_src, [('source_file', 'in_file')]),
-        (inputnode, itk2lta, [('source_file', 'src_file'),
-                              ('t1w2fsnative_xfm', 'in_file')]),
-        (get_fsnative, itk2lta, [('brain', 'dst_file')]),  # InfantFS: Use brain instead of T1
-        (inputnode, sampler, [('subjects_dir', 'subjects_dir'),
-                              ('subject_id', 'subject_id')]),
-        (itersource, targets, [('target', 'space')]),
-        (itersource, rename_src, [('target', 'subject')]),
-        (itk2lta, sampler, [('out', 'reg_file')]),
-        (targets, sampler, [('out', 'target_subject')]),
-        (rename_src, sampler, [('out_file', 'source_file')]),
-        (update_metadata, outputnode, [('out_file', 'surfaces')]),
-        (itersource, outputnode, [('target', 'target')]),
-    ])
+    workflow = connect_bold_surf_wf(
+        workflow, inputnode, outputnode, get_fsnative, itersource, targets,
+        rename_src, sampler, itk2lta, update_metadata, apply_goodvx_or_rename
+    )
     # fmt: on
+    return apply_med_surf_nans_if(
+        medial_surface_nan, workflow, inputnode, sampler, update_metadata, medial_nans
+    )
 
-    if not medial_surface_nan:
-        workflow.connect(sampler, "out_file", update_metadata, "in_file")
-        return workflow
 
-    from niworkflows.interfaces.freesurfer import MedialNaNs
+def make_ribbon_file(workflow, mem_gb, inputnode):
```

Comment:
This seems like an anatomical workflow, so it should go in the https://github.com/nipreps/nibabies/tree/master/nibabies/workflows/anatomical module.

Also, to stay consistent with nipreps best practices, let's rename/restructure this as a separate workflow.

```suggestion
def init_ribbon_volume_wf(mem_gb, name='ribbon_volume_wf'):
```